geometry.hpp

```diff
@@ -62,6 +62,10 @@ class Vector3d{
     return Vector3d(v[0]*d, v[1]*d, v[2]*d);
   }
 
+  double at( int i ) const {
+    return v[i];
+  }
+
```

Comment:
I would vote to remove this function and stick with the existing `myvector.v[1]` syntax, for several reasons:

* The member `v` is public explicitly to support coordinate access.
* This new function seems to be used very little in this PR, and it's used nowhere else in the program.
* The name `.at()` suggests a parity with `std::vector::at()`, which has explicit bounds checking.  Unlike the `std::vector` function, which throws an exception on bound overruns, his function will crash if you give it a number that's less than 0 or greater than 2.

If this function must remain, then its argument should be unsigned.